hls4ml/model/hls_layers.py

```diff
@@ -1810,6 +1814,614 @@ def _get_transforms_config(self, params):
 
         params['sublayer_configs'] = '\n'.join(sublayer_configs)
 
+class GraphBlock(Layer): #parent class for EdgeBlock, NodeBlock
+    def add_weights(self, quantizer=None, compression=False):
+        linear_count = 0
+
+        for name, module in self.submodules.items():
+            if module.__class__.__name__ == 'Linear':
+                data = self.model.get_weights_data(self.name, name, 'kernel').transpose()
+                var_name = f"{self.name}_w{linear_count}"
+                self.add_weights_variable(name=var_name, var_name=var_name, data=data, quantizer=quantizer,
+                                              compression=compression)
+                linear_count += 1
+
+        # DUMMIES
```

Comment:
Why are these weights added twice with different names?

Reply:
Are you reffering to the "add_weight" vs "add_bias" functions?

Reply:
I'm referring to these two lines:

https://github.com/Duchstf/hls4ml/pull/4/files/0318f2843be7f9c6a9009c57ffcb134ecd41aec1#diff-596f42295045305371ca3a443fce3a9c78302f2462ded2f2b16e2c3791944e95R1825

https://github.com/Duchstf/hls4ml/pull/4/files/0318f2843be7f9c6a9009c57ffcb134ecd41aec1#diff-596f42295045305371ca3a443fce3a9c78302f2462ded2f2b16e2c3791944e95R1832